### PR TITLE
fix #8761

### LINKED ISSUE #8761: [BUG] Builder does not get correct tool while deconstructing or digging.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.19

### MineColonies Version

1.19.2-1.0.1106

### Structurize Version

1.19.2-1.0.460

### Related Mods and their Versions

1. Forge Version: 43.1.47

This is run in a modpack with other mods installed. Full List [here](https://gist.github.com/theOnlyHorst/2c8aebcb720102103eea874e73ecb78d)

### Current Behavior

When deconstructing or digging the builder tries to break a block but doesn't have the appropriate tool in inventory. They try to break the block, fail and then path away a bit only to try to path back again and fail again, repeating this over and over.
Sometimes their behaviour is as expected and sometimes it isn't.

https://user-images.githubusercontent.com/34474380/201180161-23242a8a-55eb-439a-b489-baaa693d80c2.mp4

### Expected Behavior

The builder after noticing they do not have the appropriate tool in inventory, will try to get it from their hut block or request it from the request system.

### Reproduction Steps

1. Deconstruct a building or use the dig schematic with the builder on blocks that require any sort of tool to break.
2. Provide the builder with the needed tools.
3. Take those tools away before they are trying to break a block.

This does also occur should the builder put away their tools before trying to break the block.

### Logs

https://gist.github.com/theOnlyHorst/4ef9dde884983fb81ec4325398795eb9

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #8763: [BUG] Female nordic researcher model broken
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.19

### MineColonies Version

1.19.2-1.0.1106

### Structurize Version

1.19.2-1.0.459

### Related Mods and their Versions

Forge: 43.1.47
Domum: 1.19-1.0.74
Milti piston: 1.19.2-1.2.19
Block ui: 1.19-0.0.64

### Current Behavior

Model is broken,  seems to be using both model types (slim and normal)

### Expected Behavior

Not a broken model, not using both model version

### Reproduction Steps

hire a female researcher while using til nordic citizen style

### Logs

https://gist.github.com/EmmiPigen/87b84992a35ff4c853e5a06efd115c5f

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #8766: Option alwaysrendernametag does not hide guests of colony
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.19

### MineColonies Version

minecolonies-1.19.2-1.0.1106-ALPHA.jar

### Structurize Version

structurize-1.19.2-1.0.452-ALPHA.jar

### Related Mods and their Versions

Forge: forge-1.19.2-43.1.47-installer.jar

### Current Behavior

If set alwaysrendernametag to "false" and restart Minecraft then tags for citizens are disappeared but icons of colony guests are still visible even behind walls. See screenshot.

### Expected Behavior

Looks like tags visible behind walls make great effect to FPS so hide should work and may be = false by default.

### Reproduction Steps

-

### Logs

-

### Anything else?

-

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #8761
Closes #8766
Closes #8763

# Changes proposed in this pull request:
-
- Fix stuck builder
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java [hunk 1]
```diff
@@ -202,8 +208,9 @@ private void requestMaterialsState()
     @Override
     protected IAIState waitForRequests()
     {
-        if (job.hasWorkOrder() && building.getNeededResources().isEmpty())
+        if (job.hasWorkOrder() && building.getNeededResources().isEmpty() && !recalculated)
         {
+            recalculated = true;
```
Comment (uecasm) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java]: Shouldn't it also set this to false when it completes the job or something?

Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java:213]: I think so yes

Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java]: I think so yes

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java]: No, this is specifically if the builder is unloaded mid build. They have a request open and then don't recalc the list, if the builder is unloaded again, the variable auto resets to false

File: src/main/java/com/minecolonies/coremod/entity/citizen/VisitorCitizen.java [hunk 2]
```diff
@@ -743,7 +740,6 @@ protected void dropEquipment()
         }
     }
 
-    // TODO:REMOVE DEBUG
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/citizen/VisitorCitizen.java:746]: while I think its fine to remove this, please remove the entire overriden function here, all of it is debug code :D

**Review by someaddons — APPROVED** [review]

**Pull request merged**